### PR TITLE
:ghost: Upgrade Axios

### PR DESCRIPTION
**Pull request by ibolton336**
Upgrade Axios

Resolves https://issues.redhat.com/browse/MTA-2326

### CODE REVIEW

**Review by sjd78 — COMMENTED** [review]
Smoke test passed.  :smile:

File: client/src/app/api/rest.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-// hub OpenAPI definition: https://github.com/konveyor/tackle2-hub/blob/main/docs/openapi3.json
-
-import axios, { AxiosPromise } from "axios";
+import axios, {
+  AxiosHeaders,
```
Comment (rszwajko) [inline client/src/app/api/rest.ts:2]: unused

**Review by rszwajko — CHANGES_REQUESTED** [review]
We need default handler that would silence the errors - sth similar to `mock.onAny().reply(200, [])`.

**Review by rszwajko — APPROVED** [review]
@ibolton336 
I'm OK with fixing the logs in a follow-up

File: client/src/app/pages/applications/components/application-form/__tests__/application-form.test.tsx [hunk 2]
```diff
@@ -6,26 +6,30 @@ import {
...
 
 import "@testing-library/jest-dom";
-import { BusinessService } from "@app/api/models";
 import { ApplicationFormModal } from "../application-form-modal";
+import { server } from "@mocks/server";
+import { rest } from "msw";
 
 describe("Component: application-form", () => {
   const mockChangeValue = jest.fn();
+  beforeAll(() => server.listen({ onUnhandledRequest: "warn" }));
```
Comment (sjd78) [inline client/src/app/pages/applications/components/application-form/__tests__/application-form.test.tsx:18]: Does this need and `afterAll(() => server.close())`?

File: client/src/app/pages/identities/components/identity-form/__tests__/identity-form.test.tsx [hunk 3]
```diff
@@ -6,17 +6,16 @@ import {
   fireEvent,
 } from "@app/test-config/test-utils";
 
-import { IDENTITIES } from "@app/api/rest";
-import mock from "@app/test-config/mockInstance";
-
 import { IdentityForm } from "..";
 import "@testing-library/jest-dom";
+import { server } from "@mocks/server";
```
Comment (sjd78) [inline client/src/app/pages/identities/components/identity-form/__tests__/identity-form.test.tsx:11]: Can replace: 
```js
mock.onGet(`${IDENTITIES}`).reply(200, data);
```

with
```tsx
server.use(rest.get("*", (req, res, ctx) => {
  return res(ctx.json(data));
})
```

File: package.json [hunk 4]
```diff
@@ -74,6 +74,7 @@
     "typescript": "^5.1.6"
   },
   "overrides": {
-    "follow-redirects": "^1.15.6"
+    "follow-redirects": "^1.15.6",
+    "axios": "^1.6.8"
```
Comment (sjd78) [inline package.json:78]: Is this override necessary?

**Review by sjd78 — COMMENTED** [review]
Looks pretty good.  Minor comments in a few places.

**Pull request merged**